File: src/main/java/com/twsg/lnl/kotlin/c1_dataclass/g3_lombok2/Student.java
package com.twsg.lnl.kotlin.c1_dataclass.g3_lombok2;

import lombok.AllArgsConstructor;
import lombok.Data;

/**
 * @author tonyl
 */

@Data
@AllArgsConstructor
public class Student {

    private final String id;

    private final String firstName;

    private final String lastName;

    public String getFullName() {
        return firstName + " " + lastName;
    }

}

File: src/main/java/com/twsg/lnl/kotlin/c1_dataclass/g5_kotlin2/Student.kt
package com.twsg.lnl.kotlin.c1_dataclass.g5_kotlin2

/**
 * @author tonyl
 */
data class Student(
        val id: String,
        val firstName: String,
        val lastName: String
) {

    val fullName: String
        get() = "$firstName $lastName"

}